nft-marketplace/app/ui/AccountPage.jsx

```diff
@@ -0,0 +1,140 @@
+import axios from 'axios';
+import { Card } from "./components/Card";
+import { useParams, useLocation } from "react-router-dom";
+import truncateEthAddress from "truncate-eth-address";
+import { SortOptions } from "./common/SortOptions";
+import { Select } from "./components/Fields/Select";
+import { CategoryOptions } from "./common/CategoryOptions";
+
+import {
+  marketplaceAddress
+} from '../../config'
+
+import NFTMarketplace from '../../artifacts/contracts/NFTMarketplace.sol/NFTMarketplace.json';
+import {Button} from "./components/Button";
+
+export default function MyNftsPage() {
+  const [nfts, setNfts] = useState([]);
+  const [loadingState, setLoadingState] = useState('not-loaded');
+  const [category, setCategory] = useState(CategoryOptions[0].value);
+  const [sortBy, setSortBy] = useState(SortOptions[0].value);
+  const loadMoreInitialState = 4;
+  const loadMoreIncrementSize = 4;
+  const [loadMore, setLoadMore] = useState(loadMoreInitialState);
+  const { address } = useParams();
+  const location = useLocation();
+  useEffect(() => {
+    loadNFTs()
+  }, [location, category, sortBy, loadMore]);
+
+  async function loadNFTs() {
+    /* create a generic provider and query for unsold market items */
+    const provider = new ethers.providers.JsonRpcProvider();
+    const signer = provider.getSigner();
+    const marketplaceContract = new ethers.Contract(marketplaceAddress, NFTMarketplace.abi, signer);
+    const data = await marketplaceContract.fetchAllNFTsFromUser(address);
+
+    const items = await Promise.all(data.map(async i => {
+      const tokenURI = await marketplaceContract.tokenURI(i.tokenId);
+      const meta = await axios.get(tokenURI);
+      let price = ethers.utils.formatUnits(i.price.toString(), 'ether');
+      const badge = (i.owner.toLowerCase() === address.toLowerCase() && i.owner.toLowerCase() !== marketplaceAddress.toLowerCase()) ? "owned" : "for sale";
+      let item = {
+        price,
+        badge,
+        tokenId: i.tokenId.toNumber(),
+        seller: i.seller.toLowerCase(),
+        owner: i.owner.toLowerCase(),
+        name: meta.data.name,
+        image: meta.data.image,
+        tokenURI
+      };
+
+      return item;
+    }));
+
+    let filteredItems = items;
+
+    if (category === 'owned') {
+      filteredItems = items.filter((item) => item.badge === 'owned' );
+    } else if (category === 'for-sale') {
+      filteredItems = items.filter((item) => item.badge === 'for sale' );
+    }
+
+    if (sortBy === 'oldest') {
+      filteredItems.sort((a,b) => a.tokenId - b.tokenId);
+    } else if (sortBy === 'newest') {
+      filteredItems.sort((a,b) => b.tokenId - a.tokenId);
+    } else if (sortBy === 'price-low') {
+      filteredItems.sort((a,b) => a.price - b.price);
+    } else if (sortBy === 'price-high') {
+      filteredItems.sort((a,b) => b.price - a.price);
+    }
+
+    setNfts(filteredItems);
+    setLoadingState('loaded');
+  }
```

Comment:
It would be awesome to see this inside a custom hook.

just an improvement, not mandatory